extension/persistence/relational-jdbc/src/main/java/org/apache/polaris/extension/persistence/relational/jdbc/CheckedFunction.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.extension.persistence.relational.jdbc;
+
+import java.sql.SQLException;
+
+@FunctionalInterface
+interface CheckedFunction<T, R> {
+  R apply(T t) throws SQLException;
+}
```

Comment:
To be honest, I don't think it improves the code. It feels like a case of overengineering. Personally, I still prefer the `if/else` approach, it's simple, clear, and gets the job done. I'm okay with it though.

Reply:
It seems like T is always String and R is always int?

Reply:
+1 to @eric-maynard ' comment. Also, this interface can probably be an inner class in `JdbcBasePersistenceImpl`... it is not meant for sharing, is it?

Reply:
I wanted this functional interface to be used across, in case such more use case comes hence i kept generics and moved its to its own class, if we want to address this for this particular use case then defining it within the `JdbcBasePersistenceImpl` makes sense and removing generic also make sense.